### DISCUSSION TITLE
A better name for the data type 'linked records'

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#640] [Help] [answered]
We've had some feedback around the naming of Linked Records. Some users find it confusing and have offered suggestions such as:

- Relationship
- Linked Data
- Linked Columns

If you have any suggestions or opinions on this, please comment below.

Thanks.

**Comment by mjashanks** [comment]
There will be lots of differing opinions on this, I would put this down to a training issue.

Some people will understand "Linked Records" immediately. Others won't. 

My vote would be Linked Records or Relationships

**Comment by aptkingston** [comment]
There's a few of these we need to decide - linked records, true/false and options.

For options, as detailed in [this issue](https://github.com/Budibase/budibase/issues/631) we'll move "Options" to its own type and leave "Text" for just strings. I think we're all on board with that?

For true/false as detailed in [this issue](https://github.com/Budibase/budibase/issues/629) is everyone on boad with renaming to "Yes/No"?

For linked reccords, I was thinking something along the lines of what Airtable does and calling it "Link to another table" - like a user readable short description which removes all ambiguity.

I'm happy to implement all these as part of the linked records update once we make decisions.

**Reply by joebudi** [reply]
I feel true/false is better than yes/no.

I actually like 'Column Link' - it's more actionable and in the present, if that makes sense - I'm thinking of the process of adding new columns. I think 'columns' is better than 'rows'/'records' because it is sort of what you are linking.

Or

'Data Link'.

**Reply by aptkingston** [reply]
I personally prefer True/False as well - it was just beta feedback that didn't like it but I'm happy to stick with what we have on that one.

Linked records are confusing because yes you're adding a column, but the link is always row to row / record to record. The reason I like putting the word "Table" in there is because that's how the user will think about it - I have a "students" table and I'm adding a new column called "classes" and I want to link it to my other table called "classes". The name they put for that column will usually actually be the name of another table probably. Tough one.

**Reply by mike12345567** [reply]
I like true/false too, I think yes/no has a different meaning, it can't really be used for like an on/off state without it sounding a little odd, like a checkbox isn't in a "yes" or a "no" state, that implies its a question, rather than an option.

**Comment by mike12345567** [comment]
I personally have no problem with the "linked records" name but it depends on whether or not we want the `row` and `record` to be used in place of each other throughout the app, right now the backend never refers to records, only rows and automations currently refer to everything as records. "Linked row" doesn't quite have the same meaning to me, but it might make more sense when actually looking at the backend. Relationship is also good too, but I feel you'd need to say it is a column relationship or a row relationship, technically these are relationships between rows but the link is made at a column level which is a little confusing.

**Reply by aptkingston** [reply]
Yea I was surprised that Airtable uses the terminology of "field" and "record". I guess "row" and "column" is still more accessible though if a user is coming from a background of only spreadsheets.

I think relationship would be ok by itself since we aren't going to ever have other types of relationship, and what you're linking to can be implied by what params we ask for as part of the link. Maybe we need a poll haha.

**Comment by shogunpurple** [comment]
I thought linked records was quite explicit, but "Relationships" seems like a better choice to me - it's a non-technical term and as Mike alluded to I think a little bit of documentation, training or help text would go a long way, regardless of what we call it.

What you are creating a relationship to will probably be implicit, based on the fact it will ask you to choose a column in the other table to link to.

**Comment by thejasonparker** [comment, marked as answer]
"Relationship" gets my vote as it's something I would immediately understand as it's used for databases. It's also non specific so could be used for any data links. 

On this feature  do you have an ETA for this? I understand Budibase is very early in its development cycle but without this is really stops any useful way to use it as data links are the bedrock of any application build.

**Reply by joebudi** [reply]
Relationships it is!

I believe @aptkingston and @mike12345567 are adding the finishing touches to 'Relationships' so it should be ready for the next release (next week). I personally saw a demo yesterday, and the Relationships feature looks great.
